### ISSUE TITLE
Make task processing crash safe

### ISSUE THREAD

**Issue by iclems** [googlearchive/firebase-work-queue#3]
Love this example ;-) I think the item should not be nullified on Firebase directly when the job is taken by one worker, but should be rather marked as "taken", processed, and removed once the processing is done. Otherwise, if your server crashes during the processing phase, you may still end up having lost a task.

**Comment by markoshust** [comment]
I needed this for a more failsafe option... just made a PR with a very simple 'status' flag that changes it to 'processing' when it is picked up from the worker (with the idea that someone else can use status for other things (error, hold, etc.), then removed from the queue like usual after the user-defined callback is processed.

It'd be a good idea to add an error callback on worker.js so we can update flag from 'processing' to 'error'. this way you can have other workers notify you immediately with a problem processing from the queue.

**Comment by markoshust** [comment]
A big note here -- if you use this, the queue will get stuck on any queue item with a 'status'. " on any error and kinda get stuck. I expanded this script to create some additional functionality including when the queue items were created, status on queue items, error handling and safety in case of something like a crashed server.

I updated this PR, but understand if this is too much to merge into this repo as it changes the functionality quite a bit, so just kill the PR and reference by repo if you want all this and don't want to merge.
https://github.com/markoshust/firebase-work-queue-advanced
